### PR TITLE
Add OneShard page

### PR DESCRIPTION
**Pull request by Simran-B**

### CODE REVIEW

File: 3.6/oneshard.md [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+  5   scatter-in-cluster
+  6   distribute-filtercalc-to-cluster
+  7   distribute-sort-to-cluster
+  8   remove-unnecessary-remote-scatter
+  9   sort-limit
+ 10   move-filters-into-enumerate
+ 11   parallelize-gather
+```
+
+Without the oneshard feature all documents have potentially to be send to the
```
Comment (jsteemann) [inline 3.6/oneshard.md]:
```suggestion
Without the OneShard feature all documents have potentially to be sent to the
```

File: 3.6/oneshard.md [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+coordinator for further processing. With this simple query this is actually not
+true, because we do some super awesome optimization that is able to reduce
```
Comment (jsteemann) [inline 3.6/oneshard.md]:
```suggestion
true, because we perform some other optimization that reduces
```

File: 3.6/oneshard.md [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+the number of documents, but when the queries become more complex and more
```
Comment (jsteemann) [inline 3.6/oneshard.md]:
```suggestion
the number of documents.
```

File: 3.6/oneshard.md [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+collections are involved the optimizer will not be able to help as much.
```
Comment (jsteemann) [inline 3.6/oneshard.md]:
```suggestion
```

File: 3.6/oneshard.md [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+the final result. This can reduce resource usage and communication overhead
+for the coordinator dramatically as show in the [example]() below.
+
+
+
+How to use the OneShard feature?
+--------------------------------
+
+In order to use the OneShard feature you just need to have the ArangoDB enterprise
+edition installed. Then the optimizer will automatically make use of the of the
```
Comment (jsteemann) [inline 3.6/oneshard.md]:
```suggestion
edition installed. Then the optimizer will automatically make use of the
```

File: 3.6/oneshard.md [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+OneShard feature if the collections used in the query are eligible. To be eligible
+the single shards of all the collections in question have to be on the same dbserver.
```
Comment (jsteemann) [inline 3.6/oneshard.md]:
```suggestion
the single shards of all the collections in question have to be on the same dbserver and have to use the same sharding instructions. Internally, this is ensured by setting the `distributeShardsLike` attribute for collections properly.
```

File: 3.6/oneshard.md [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+
+What is the OneShard feature?
+-----------------------------
+
+Under certain conditions big parts of the execution can be pushed to a single
+db-servers. This can result in big performance gains because usually multiple
+db-servers and a coordinator are involved in the execution of a query.
+
+Without the OneShard feature query processing works as follows. The coordinator
+accepts and analyses the query. If collections are accessed the coordinator
```
Comment (jsteemann) [inline 3.6/oneshard.md]:
```suggestion
accepts and analyzes the query. If collections are accessed the coordinator
```

File: 3.6/oneshard.md [hunk 8]
```diff
@@ -0,0 +1,203 @@
...
+distributes the accesses to collections to different db-servers that hold parts
+(shards) of the collections in question.
+This distributed access requires network-traffic from coordinator to dbservers
```
Comment (jsteemann) [inline 3.6/oneshard.md]: `dbservers` vs. `db-servers`. There should be a consistent spelling.

File: 3.6/oneshard.md [hunk 9]
```diff
@@ -0,0 +1,203 @@
...
+and back from dbserves to coordinators and is therefore expensive. Another cost
+factor is the memory and cpu-time required on the coordinator, when it has to
+process several concurrent complex queries. In such situations coordinators
+might become a bottleneck in query processing. Because they need to send and
+receive data on several connections, build up results for collection accesses
+from the received parts followed by further processing.
+If the one shard feature can be used for a query then the whole query is pushed
+to a db-server and executed on the server. The coordinator will only get back
+the final result. This can reduce resource usage and communication overhead
+for the coordinator dramatically as show in the [example]() below.
```
Comment (jsteemann) [inline 3.6/oneshard.md]:
```suggestion
for the coordinator dramatically as shown in the [example]() below.
```

File: 3.6/oneshard.md [hunk 10]
```diff
@@ -0,0 +1,203 @@
...
+description: queries using the OneShard feature can be faster
+---
+
+OneShard enterpise-feature
+==========================
+
+What is the OneShard feature?
+-----------------------------
+
+Under certain conditions big parts of the execution can be pushed to a single
```
Comment (Simran-B) [inline 3.6/oneshard.md]:
```suggestion
Under certain conditions large parts of the execution can be pushed to a single
```

File: 3.6/oneshard.md [hunk 11]
```diff
@@ -0,0 +1,203 @@
+---
+layout: default
+description: queries using the OneShard feature can be faster
+---
+
+OneShard enterpise-feature
```
Comment (Simran-B) [inline 3.6/oneshard.md]:
```suggestion
OneShard Cluster Deployment
```

File: 3.6/oneshard.md [hunk 12]
```diff
@@ -0,0 +1,203 @@
...
+==========================
```
Comment (Simran-B) [inline 3.6/oneshard.md]:
```suggestion
==========================

{% hint 'info' %}
This option is only available in the
[**Enterprise Edition**](https://www.arangodb.com/why-arangodb/arangodb-enterprise/){:target="_blank"},
also available as [**managed service**](https://www.arangodb.com/managed-service/){:target="_blank"}.
{% endhint %}
```

File: 3.6/oneshard.md [hunk 13]
```diff
@@ -0,0 +1,203 @@
...
+
+What is the OneShard feature?
+-----------------------------
+
+Under certain conditions big parts of the execution can be pushed to a single
+db-servers. This can result in big performance gains because usually multiple
```
Comment (Simran-B) [inline 3.6/oneshard.md]:
```suggestion
DB-Servers. This can result in performance gains because usually multiple
```

File: 3.6/oneshard.md [hunk 14]
```diff
@@ -0,0 +1,203 @@
...
+db-servers and a coordinator are involved in the execution of a query.
```
Comment (Simran-B) [inline 3.6/oneshard.md]:
```suggestion
DB-Servers (and a Coordinator) are involved in the execution of a query.
```

File: 3.6/oneshard.md [hunk 15]
```diff
@@ -0,0 +1,203 @@
...
+
+Without the OneShard feature query processing works as follows. The coordinator
```
Comment (Simran-B) [inline 3.6/oneshard.md]:
```suggestion
Without the OneShard feature query processing works as follows: The Coordinator
```

File: 3.6/oneshard.md (hunk 7 above)
Comment (Simran-B) [inline 3.6/oneshard.md]:
```suggestion
accepts and analyses the query. If collections are accessed the Coordinator
```

File: 3.6/oneshard.md [hunk 16]
```diff
@@ -0,0 +1,203 @@
...
+accepts and analyses the query. If collections are accessed the coordinator
+distributes the accesses to collections to different db-servers that hold parts
```
Comment (Simran-B) [inline 3.6/oneshard.md]:
```suggestion
distributes the accesses to collections to different DB-Servers that hold parts
```

File: 3.6/oneshard.md (hunk 8 above)
Comment (Simran-B) [reply 3.6/oneshard.md]: Yes, and I think we generally use `DBServers`. I think it would make sense to change this to `DB-Servers` however.

**Comment by ObiWahn** [comment]
Thank you for doing all my work, but would have done it. I just did not know that the PR was created.

File: 3.6/oneshard.md [hunk 17]
```diff
@@ -0,0 +1,221 @@
...
+because we specified the `{ "sharding": "single" }` options object when
+creating the database. To do this manually one would create a collection in
+the following way:
+
+```js
+db._create("example2", {"numberOfShards":1 , "distributeShardsLike":"_graphs"})
+```
+
+Here we used again the `_graphs` collection, but any other existing
+collection, that has not been created with the `distributesShardsLike`
```
Comment (jsteemann) [inline 3.6/oneshard.md]:
```suggestion
collection, that has not been created with the `distributeShardsLike`
```

File: 3.6/oneshard.md [hunk 18]
```diff
@@ -0,0 +1,221 @@
...
+option could have been used here.
+
+### Running Queries
+
+First we insert a few documents into a collection, create a query and
+explain what will be done.
+
+```js
+127.0.0.1:8529@oneShardDB> for(let i=0; i<10000; i++) { db.example.insert( {"value":i} ) }
```
Comment (jsteemann) [inline 3.6/oneshard.md]:
```suggestion
127.0.0.1:8529@oneShardDB> for (let i = 0; i < 10000; i++) { db.example.insert({ "value" : i }); }
```

File: 3.6/oneshard.md [hunk 19]
```diff
@@ -0,0 +1,221 @@
...
+{
+  "_id" : "example/6010134",
+  "_key" : "6010134",
+  "_rev" : "_ZrqaOt2---"
+}
```
Comment (jsteemann) [inline 3.6/oneshard.md]: not relevant and can be removed here

File: 3.6/oneshard.md [hunk 20]
```diff
@@ -0,0 +1,221 @@
...
+
+127.0.0.1:8529@oneShardDB> q="FOR doc IN @@collection FILTER doc.value % 2 == 0 SORT doc.value ASC LIMIT 10 RETURN doc"
```
Comment (jsteemann) [inline 3.6/oneshard.md]:
```suggestion
127.0.0.1:8529@oneShardDB> q = "FOR doc IN @@collection FILTER doc.value % 2 == 0 SORT doc.value ASC LIMIT 10 RETURN doc";
```

File: 3.6/oneshard.md [hunk 21]
```diff
@@ -0,0 +1,221 @@
...
+FOR doc IN @@collection FILTER doc.value % 2 == 0 SORT doc.value ASC LIMIT 10 RETURN doc
+
+127.0.0.1:8529@oneShardDB> db._explain(q, { "@collection":"example" })
```
Comment (jsteemann) [inline 3.6/oneshard.md]:
```suggestion
127.0.0.1:8529@oneShardDB> db._explain(q, { "@collection" : "example" })
```

File: 3.6/oneshard.md [hunk 22]
```diff
@@ -0,0 +1,221 @@
...
+As it can be seen in the explain output almost the complete query is
+executed on the DB-Server (`DBS` for nodes 1-7) and only 10 documents are
+transferred to the Coordinator. In case we do the same with a collection
+that consists of several shards we get a different result:
+
+```js
+127.0.0.1:8529@_system> db._create("example", { numberOfShards : 5})
+[ArangoCollection 6863477, "example" (type document, status loaded)]
+
+127.0.0.1:8529@_system> for(let i=0; i<10000; i++) { db.example.insert( {"value":i} ) }
```
Comment (jsteemann) [inline 3.6/oneshard.md]:
```suggestion
127.0.0.1:8529@_system> for (let i = 0; i < 10000; i++) { db.example.insert({ "value" : i }); }
```

File: 3.6/oneshard.md [hunk 23]
```diff
@@ -0,0 +1,221 @@
...
+{
+  "_id" : "example/6873482",
+  "_key" : "6873482",
+  "_rev" : "_ZrrC_0K---"
+}
```
Comment (jsteemann) [inline 3.6/oneshard.md]: not relevant and should be removed here

File: 3.6/oneshard.md [hunk 24]
```diff
@@ -0,0 +1,221 @@
...
+
+127.0.0.1:8529@_system> db._explain(q, { "@collection":"example" })
```
Comment (jsteemann) [inline 3.6/oneshard.md]:
```suggestion
127.0.0.1:8529@_system> db._explain(q, { "@collection" : "example" })
```

**Pull request merged**